Fail: Azure unit tests TestCreateVolume and TestDeleteVolume fail with master golang

### Which jobs are failing?

Our internal CI job that runs k8s unit tests with master golang.

### Which tests are failing?

k8s.io/kubernetes/vendor/k8s.io/legacy-cloud-providers/azure: TestCreateVolume
k8s.io/kubernetes/vendor/k8s.io/legacy-cloud-providers/azure: TestDeleteVolume

### Since when has it been failing?

Probably after the master golang GITCOMMIT=4a0a2b33dfa3c99250efa222439f2c27d6780e4a
Change: https://go-review.googlesource.com/c/go/+/432898

### Testgrid link

_No response_

### Reason for failure (if possible)

_No response_

### Anything else we need to know?

```
k8s.io/kubernetes/vendor/k8s.io/legacy-cloud-providers/azure: TestCreateVolume 
{Failed  === RUN   TestCreateVolume
    azure_blobDiskController_test.go:91: 
        	Error Trace:	/home/prow/go/src/github.com/kubernetes/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/legacy-cloud-providers/azure/azure_blobDiskController_test.go:91
        	Error:      	Not equal: 
        	            	expected: *fmt.wrapError(&fmt.wrapError{msg:"could not get storage key for storage account testsa: could not get storage key for storage account testsa: Retriable: false, RetryAfter: 0s, HTTPStatusCode: 500, RawError: %!w(<nil>)", err:error(nil)})
        	            	actual  : *errors.errorString(&errors.errorString{s:"could not get storage key for storage account testsa: could not get storage key for storage account testsa: Retriable: false, RetryAfter: 0s, HTTPStatusCode: 500, RawError: %!w(<nil>)"})
        	Test:       	TestCreateVolume
--- FAIL: TestCreateVolume (1.37s)
}
```
```
k8s.io/kubernetes/vendor/k8s.io/legacy-cloud-providers/azure: TestDeleteVolume
{Failed  === RUN   TestDeleteVolume
    azure_blobDiskController_test.go:127: 
        	Error Trace:	/home/prow/go/src/github.com/kubernetes/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/legacy-cloud-providers/azure/azure_blobDiskController_test.go:127
        	Error:      	Not equal: 
        	            	expected: *fmt.wrapError(&fmt.wrapError{msg:"no key for storage account foo, err Retriable: false, RetryAfter: 0s, HTTPStatusCode: 500, RawError: %!w(<nil>)", err:error(nil)})
        	            	actual  : *errors.errorString(&errors.errorString{s:"no key for storage account foo, err Retriable: false, RetryAfter: 0s, HTTPStatusCode: 500, RawError: %!w(<nil>)"})
        	Test:       	TestDeleteVolume
    azure_blobDiskController_test.go:130: 
        	Error Trace:	/home/prow/go/src/github.com/kubernetes/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/legacy-cloud-providers/azure/azure_blobDiskController_test.go:130
        	Error:      	Not equal: 
        	            	expected: *fmt.wrapError(&fmt.wrapError{msg:"no key for storage account foo, err Retriable: false, RetryAfter: 0s, HTTPStatusCode: 500, RawError: %!w(<nil>)", err:error(nil)})
        	            	actual  : *errors.errorString(&errors.errorString{s:"no key for storage account foo, err Retriable: false, RetryAfter: 0s, HTTPStatusCode: 500, RawError: %!w(<nil>)"})
        	Test:       	TestDeleteVolume
W0930 01:23:00.203322   88414 azure_blobDiskController.go:131] azureDisk - failed to delete blob https://foo.blob./vhds/bar.vhd err: storage: service returned error: StatusCode=403, ErrorCode=AccountIsDisabled, ErrorMessage=The specified account is disabled.
RequestId:246afb77-401e-0005-706b-d485ea000000
Time:2022-09-30T01:23:00.1710204Z, RequestInitiated=Fri, 30 Sep 2022 01:22:59 GMT, RequestId=246afb77-401e-0005-706b-d485ea000000, API Version=, QueryParameterName=, QueryParameterValue=
--- FAIL: TestDeleteVolume (0.47s)
}
```

### Relevant SIG(s)

/sig testing

--- a/staging/src/k8s.io/legacy-cloud-providers/azure/azure_blobDiskController_test.go
+++ b/staging/src/k8s.io/legacy-cloud-providers/azure/azure_blobDiskController_test.go
@@ -88,7 +88,7 @@ func TestCreateVolume(t *testing.T) {
 	diskName, diskURI, requestGB, err := b.CreateVolume("testBlob", "testsa", "type", b.common.location, 10)
 	expectedErr := fmt.Errorf("could not get storage key for storage account testsa: could not get storage key for "+
 		"storage account testsa: Retriable: false, RetryAfter: 0s, HTTPStatusCode: 500, RawError: %w", error(nil))
-	assert.Equal(t, expectedErr, err)
+	assert.EqualError(t, err, expectedErr.Error())
 	assert.Empty(t, diskName)
 	assert.Empty(t, diskURI)
 	assert.Zero(t, requestGB)
@@ -124,10 +124,10 @@ func TestDeleteVolume(t *testing.T) {
 	diskURL := "https://foo.blob./vhds/bar.vhd"
 	err := b.DeleteVolume(diskURL)
 	expectedErr := fmt.Errorf("no key for storage account foo, err Retriable: false, RetryAfter: 0s, HTTPStatusCode: 500, RawError: %w", error(nil))
-	assert.Equal(t, expectedErr, err)
+	assert.EqualError(t, err, expectedErr.Error())
 
 	err = b.DeleteVolume(diskURL)
-	assert.Equal(t, expectedErr, err)
+	assert.EqualError(t, err, expectedErr.Error())
 
 	mockSAClient.EXPECT().ListKeys(gomock.Any(), b.common.resourceGroup, "foo").Return(storage.AccountListKeysResult{
 		Keys: &[]storage.AccountKey{

